postgresql/migrations/000010_notification_subscription.up.sql

```diff
@@ -0,0 +1,7 @@
+CREATE TABLE public.notification_subscriptions (
+    id bigserial primary key,
+	pubkey bytea NOT NULL,
```

Comment:
pubkey, url should be unique if we want to prevent duplicate registrations.
Also I think it worth adding timestamp (updated_at) so we can implement a policy in the Store to retrieve only the non-stale registrations (for example those that were updated in the last week/month).

Reply:
> 1. I was thinking it could be useful to keep stale registrations around later, because I want to add a notification log in the database to troubleshoot them. That's usually a good practice for webhooks in enterprise type systems. We could also log a line in a logfile instead if we don't want to keep that data. What do you think?

Agree, we don't have to implement a policy now.



> 2\. Do you think we should only allow a single webhook registration at a time per pubkey? Regardless of the URL used? And if so, would a new registration overwrite the old one? Or do we want to return an error if the registration is already there and let the client explicitly remove the registration?

From multi app perspective we will need to support several registrations, we might need to limit it to some number in the future. but for now I wouldn't limit it.

Reply:
> > 1. I was thinking it could be useful to keep stale registrations around later, because I want to add a notification log in the database to troubleshoot them. That's usually a good practice for webhooks in enterprise type systems. We could also log a line in a logfile instead if we don't want to keep that data. What do you think?
> 
> Agree, we don't have to implement a policy now.
This record can be considered as not stale unless there no more channels from this nodeid to the lsp. I suggest to delete the record when all the channels from the lsp to this nodeid are (force)closed
> 
> > 2. Do you think we should only allow a single webhook registration at a time per pubkey? Regardless of the URL used? And if so, would a new registration overwrite the old one? Or do we want to return an error if the registration is already there and let the client explicitly remove the registration?
> 
> From multi app perspective we will need to support several registrations, we might need to limit it to some number in the future. but for now I wouldn't limit it.

So the url has to be an array of urls (I prefer a jsonb field which is more portable)

Reply:
> So the url has to be an array of urls (I prefer a jsonb field which is more portable)

That's the way it's currently implemented. That's why there is no unique constraint on anything there currently. I prefer using multiple rows per pubkey rather than putting a json column and have a unique constraint on the pubkey. There's an index on the pubkey, so lookups should be really quick.

Reading above comments I think for this initial version we can add a `created_at` timestamp. Then in a later PR I can implement a policy for removing notification subscriptions.

Reply:
@roeierez Can you review this again? It has a created_at and updated_at column now and a unique constraint on the pubkey, url combination.